### ISSUE TITLE
RuntimeError: mat1 and mat2 shapes cannot be multiplied (282x577 and 201x192)

### ISSUE THREAD

**Issue by morganne00** [marcoppasini/MelGAN-VC#19]
Hi, 
Thanks for sharing this notebook. I'm running the notebook as it is, and run into some issues. 
First, running the spectrogram cell with Pytorch gives a warning: 
`UserWarning: At least one mel filterbank has all zero values. The value for `n_mels` (192) may be set too high. Or, the value for `n_freqs` (201) may be set too low.`

When I continue, and reach the "#Generating Mel-Spectrogram dataset" cell, I get this error:
```
RuntimeError                              Traceback (most recent call last)
/tmp/ipykernel_30967/1562027017.py in <module>
      6 #MALE1
      7 awv = audio_array('cmu_us_clb_arctic/wav')                               #get waveform array from folder containing wav files
----> 8 aspec = tospec(awv)                                                                 #get spectrogram array
      9 adata = splitcut(aspec)                                                             #split spectrogams to fixed length
     10 #FEMALE1

/tmp/ipykernel_30967/959824048.py in tospec(data)
      6   for i in range(data.shape[0]):
      7     x = data[i]
----> 8     S=prep(x)
      9     S = np.array(S, dtype=np.float32)
     10     specs[i]=np.expand_dims(S, -1)

/tmp/ipykernel_30967/2916653096.py in prep(wv, hop)
     79 
     80 def prep(wv,hop=192):
---> 81   S = np.array(torch.squeeze(melspecfunc(torch.Tensor(wv).view(1,-1))).detach().cpu())
     82   S = librosa.power_to_db(S)-ref_level_db
     83   return normalize(S)

/tmp/ipykernel_30967/2916653096.py in melspecfunc(waveform)
     26 def melspecfunc(waveform):
     27   specgram = specfunc(waveform)
---> 28   mel_specgram = melfunc(specgram)
     29   return mel_specgram
     30 

~/venv/pytorch/lib/python3.8/site-packages/torchaudio/transforms.py in forward(self, specgram)
    394 
    395         # (..., time, freq) dot (freq, n_mels) -> (..., n_mels, time)
--> 396         mel_specgram = torch.matmul(specgram.transpose(-1, -2), self.fb).transpose(-1, -2)
    397 
    398         return mel_specgram

RuntimeError: mat1 and mat2 shapes cannot be multiplied (282x577 and 201x192)
```
I would appreciate any help to get this work.

**Comment by marcoppasini** [comment]
Hi @morganne00! Thank you for your interest in MelGAN-VC!

The notebook as it is right now runs smoothly on google colab with all the standard parameters. Looking at your error code, it looks like you changed one or more parameters in the notebook. Try reopening the original version and try running it on colab, it should work!

**Comment by morganne00** [comment]
Hi @marcoppasini ! Thank you for your reply.

Running the notebook on Google Colab without any changes, I run into problems already in the beginning, downloading and extracting the datasets: 
```
--2022-04-17 05:03:55--  http://festvox.org/cmu_arctic/cmu_arctic/packed/cmu_us_bdl_arctic-0.95-release.zip
Resolving festvox.org (festvox.org)... 199.4.150.153
Connecting to festvox.org (festvox.org)|199.4.150.153|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-04-17 05:03:55 ERROR 404: Not Found.

[cmu_us_bdl_arctic-0.95-release.zip]
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of cmu_us_bdl_arctic-0.95-release.zip or
        cmu_us_bdl_arctic-0.95-release.zip.zip, and cannot find cmu_us_bdl_arctic-0.95-release.zip.ZIP, period.
[...]
```
The zip files are downloaded, but not extracted. 

So I have run the notebook locally, and noticed that the datasets are downloaded and extracted to the main directory where the notebook is. But the original code looks for them in `../content/ `.
So running the code as it is, this error is generated:
```
ValueError                                Traceback (most recent call last)

/tmp/ipykernel_30967/2226662345.py in <module>
      5   return tf.image.random_crop(x, size=[hop, 3*shape, 1])
      6 
----> 7 dsa = tf.data.Dataset.from_tensor_slices(adata).repeat(50).map(proc, num_parallel_calls=tf.data.experimental.AUTOTUNE).shuffle(10000).batch(bs, drop_remainder=True)
      8 dsb = tf.data.Dataset.from_tensor_slices(bdata).repeat(50).map(proc, num_parallel_calls=tf.data.experimental.AUTOTUNE).shuffle(10000).batch(bs, drop_remainder=True)

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/data/ops/dataset_ops.py in map(self, map_func, num_parallel_calls, deterministic, name)
   2004       return MapDataset(self, map_func, preserve_cardinality=True, name=name)
   2005     else:
-> 2006       return ParallelMapDataset(
   2007           self,
   2008           map_func,

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/data/ops/dataset_ops.py in __init__(self, input_dataset, map_func, num_parallel_calls, deterministic, use_inter_op_parallelism, preserve_cardinality, use_legacy_function, name)
   5499     self._input_dataset = input_dataset
   5500     self._use_inter_op_parallelism = use_inter_op_parallelism
-> 5501     self._map_func = StructuredFunctionWrapper(
   5502         map_func,
   5503         self._transformation_name(),

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/data/ops/dataset_ops.py in __init__(self, func, transformation_name, dataset, input_classes, input_shapes, input_types, input_structure, add_to_graph, use_legacy_function, defun_kwargs)
   4531         fn_factory = trace_tf_function(defun_kwargs)
   4532 
-> 4533     self._function = fn_factory()
   4534     # There is no graph to add in eager mode.
   4535     add_to_graph &= not context.executing_eagerly()

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/eager/function.py in get_concrete_function(self, *args, **kwargs)
   3242          or `tf.Tensor` or `tf.TensorSpec`.
   3243     """
-> 3244     graph_function = self._get_concrete_function_garbage_collected(
   3245         *args, **kwargs)
   3246     graph_function._garbage_collector.release()  # pylint: disable=protected-access

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/eager/function.py in _get_concrete_function_garbage_collected(self, *args, **kwargs)
   3208       args, kwargs = None, None
   3209     with self._lock:
-> 3210       graph_function, _ = self._maybe_define_function(args, kwargs)
   3211       seen_names = set()
   3212       captured = object_identity.ObjectIdentitySet(

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/eager/function.py in _maybe_define_function(self, args, kwargs)
   3555 
   3556           self._function_cache.missed.add(call_context_key)
-> 3557           graph_function = self._create_graph_function(args, kwargs)
   3558           self._function_cache.primary[cache_key] = graph_function
   3559 

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/eager/function.py in _create_graph_function(self, args, kwargs, override_flat_arg_shapes)
   3390     arg_names = base_arg_names + missing_arg_names
   3391     graph_function = ConcreteFunction(
-> 3392         func_graph_module.func_graph_from_py_func(
   3393             self._name,
   3394             self._python_function,

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/framework/func_graph.py in func_graph_from_py_func(name, python_func, args, kwargs, signature, func_graph, autograph, autograph_options, add_control_dependencies, arg_names, op_return_value, collections, capture_by_value, override_flat_arg_shapes, acd_record_initial_resource_uses)
   1141         _, original_func = tf_decorator.unwrap(python_func)
   1142 
-> 1143       func_outputs = python_func(*func_args, **func_kwargs)
   1144 
   1145       # invariant: `func_outputs` contains only Tensors, CompositeTensors,

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/data/ops/dataset_ops.py in wrapped_fn(*args)
   4508           attributes=defun_kwargs)
   4509       def wrapped_fn(*args):  # pylint: disable=missing-docstring
-> 4510         ret = wrapper_helper(*args)
   4511         ret = structure.to_tensor_list(self._output_structure, ret)
   4512         return [ops.convert_to_tensor(t) for t in ret]

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/data/ops/dataset_ops.py in wrapper_helper(*args)
   4438       if not _should_unpack(nested_args):
   4439         nested_args = (nested_args,)
-> 4440       ret = autograph.tf_convert(self._func, ag_ctx)(*nested_args)
   4441       if _should_pack(ret):
   4442         ret = tuple(ret)

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/util/traceback_utils.py in error_handler(*args, **kwargs)
    151     except Exception as e:
    152       filtered_tb = _process_traceback_frames(e.__traceback__)
--> 153       raise e.with_traceback(filtered_tb) from None
    154     finally:
    155       del filtered_tb

~/venv/sound/lib/python3.8/site-packages/tensorflow/python/framework/func_graph.py in autograph_handler(*args, **kwargs)
   1127           except Exception as e:  # pylint:disable=broad-except
   1128             if hasattr(e, "ag_error_metadata"):
-> 1129               raise e.ag_error_metadata.to_exception(e)
   1130             else:
   1131               raise

ValueError: in user code:

    File "/tmp/ipykernel_30967/2226662345.py", line 5, in proc  *
        return tf.image.random_crop(x, size=[hop, 3*shape, 1])

    ValueError: Dimensions must be equal, but are 0 and 3 for '{{node random_crop/GreaterEqual}} = GreaterEqual[T=DT_INT32](random_crop/Shape, random_crop/size)' with input shapes: [0], [3].
```
This is because no .wav files are found in "../content/cmu_us_clb_arctic/wav" and `audio_array` is empty
I simply moved the dataset directories into dataset/ and changed the path in the code accordingly. Now the .wav files are found, but there appears to be a problem with matmtul with different matrix sizes.

**Comment by marcoppasini** [comment]
What a coincidence! I ran the whole notebook before writing the last message and everything executed smoothly! I tried again today and the download links seemed to be broken.
I updated the notebook with the new links and ran a few training iterations, everything works now!
I do not know where your shape mismatch error comes from, especially if you did not change anything in the notebook. Try again with the new notebook and hopefully everything will work. And with regards to the path, '../content' is the base path of colab, so no problems there.

Thank you for letting me know about the broken links!
